src/test/java/picard/sam/RevertSamTest.java

```diff
@@ -592,4 +604,135 @@ public void testHardClipRoundTrip() throws Exception {
             }
         }
     }
+
+
+    private final static boolean OUTPUT_BY_READ_GROUP = true;
+    // LOCAL_OUTPUT_MAP means the output map file itself is stored locally. The output SAM files will always be in the cloud.
+    private enum OutputMapOptions { LOCAL_OUTPUT_MAP, CLOUD_OUTPUT_MAP, NO_OUTPUT_MAP }
+
+    @DataProvider(name = "cloudTestData")
+    public Object[][] getCloudTestData() {
+        final PicardHtsPath tempCramOutputPath = PicardBucketUtils.getTempFilePath(GCloudTestUtils.TEST_OUTPUT_DEFAULT_STR, "RevertSam", ".cram");
+        return new Object[][]{
+                // Output by read group without the output map, write output bams in the cloud
+                // Since the output file names are determined within RevertSam, we can't designate the output files to be temp files.
+                // So we mark these files to be deleted. In order to avoid collisions, there cannot be another test case with the same BAM and output options.
+                {NA12878_MEDIUM, DEFAULT_CLOUD_TEST_OUTPUT_DIR, OUTPUT_BY_READ_GROUP, OutputMapOptions.NO_OUTPUT_MAP },
+                // Output by read group using the output map, write output bams in the cloud
+                {NA12878_MEDIUM, null, OUTPUT_BY_READ_GROUP, OutputMapOptions.CLOUD_OUTPUT_MAP },
+                // Output by read group using the local output map, write output bams in the cloud
+                {NA12878_MEDIUM, null, OUTPUT_BY_READ_GROUP, OutputMapOptions.LOCAL_OUTPUT_MAP },
+                // Cram input, create a cram for each read group
+                {NA12878_MEDIUM_CRAM, null, OUTPUT_BY_READ_GROUP, OutputMapOptions.CLOUD_OUTPUT_MAP },
+                // Cram input, single output cram
+                {NA12878_MEDIUM_CRAM, tempCramOutputPath, !OUTPUT_BY_READ_GROUP, OutputMapOptions.NO_OUTPUT_MAP }};
+    }
+
+    /**
+     *
+     * Write the output map for Revert Sam based on the read groups present in the header file of the given SAM file.
+     *
+     * @param sam        The read names in the header of this sam file will be used as rows
+     * @param outputBase The directory where the per read group output files will live
+     * @return The PicardHtsPath to the tsv with each row a pair (read name, output), where output points to a temp file
+o     * created in the directory specified in outputBase.
+     */
+    private PicardHtsPath getTestReadGroupMapFile(final PicardHtsPath sam, final String outputBase) {
```

Comment:
If outputBase must be a directory, I would rename it to `outputDirectory`.

Reply:
Renamed